socs/snmp.py

```diff
@@ -0,0 +1,113 @@
+    of this object.
+
+    Note: This helper currently only supports SNMPv1.
+
+    .. _SNMP Operations: http://snmplabs.com/pysnmp/docs/pysnmp-hlapi-tutorial.html
+
+    Parameters
+    ----------
+    address : str
+        Address of the SNMP Agent to send GET/SET requests to
+    port : int
+        Associated port for SNMP communication. Default is 161
+
+    Attributes
+    ----------
+    snmp_engine : pysnmp.entity.engine.SnmpEngine
+        PySNMP engine
+    address : str
+        Address of the SNMP Agent to send GET/SET requests to
+    udp_transport : pysnmp.hlapi.twisted.transport.UdpTransportTarget
+        UDP transport for UDP over IPv4
+    log : txaio.tx.Logger
+        txaio logger object
+
+    """
+
+    def __init__(self, address, port=161):
+        self.snmp_engine = SnmpEngine()
+        self.address = address
+        self.udp_transport = UdpTransportTarget((address, port))
+        self.log = txaio.make_logger()
+
+    def _success(self, args):
+        """Success callback for getCmd.
+
+        Taken from Twisted example for SNMPv1 from pySNMP documentation:
+        http://snmplabs.com/pysnmp/examples/hlapi/twisted/contents.html
+
+        Returns
+        -------
+        list
+            A sequence of ObjectType class instances representing MIB variables
+            returned in SNMP response.
+
+        """
+        (error_status, error_index, var_binds) = args
+
+        if error_status:
+            self.log.error('%s: %s at %s' % (self.address,
+                                             error_status.prettyPrint(),
+                                             error_index and
+                                             var_binds[int(error_index) - 1][0] or '?'))
+        else:
+            for var in var_binds:
+                self.log.debug(' = '.join([x.prettyPrint() for x in var]))
+
+        return var_binds
+
+    def _failure(self, error_indication):
+        """Failure Errback for getCmd.
+
+        Taken from Twisted example for SNMPv1 from pySNMP documentation:
+        http://snmplabs.com/pysnmp/examples/hlapi/twisted/contents.html
+
+        """
+        self.log.error('%s failure: %s' % (self.address, error_indication))
+
+    def get(self, oid_list):
+        """Issue a getCmd to get SNMP OID states.
+
+        Example
+        -------
+        >>> snmp = SNMPTwister('localhost', 161)
+        >>> snmp.get([ObjectType(ObjectIdentity('MBG-SNMP-LTNG-MIB',
+                                                'mbgLtNgRefclockState',
+                                                1)),
+                      ObjectType(ObjectIdentity('MBG-SNMP-LTNG-MIB',
+                                                'mbgLtNgRefclockLeapSecondDate',
+                                                1))])
+
+        Parameters
+        ----------
+        oid_list : list
+            List of high-level MIB Object OIDs. See `Specifying MIB Objects`_ for
+            more info
+
```

Comment:
How would you feel about accepting either
```
ObjectType(ObjectIdentity('MBG-SNMP-LTNG-MIB', 'mbgLtNgRefclockState', 1))
```
or simply
```
('MBG-SNMP-LTNG-MIB', 'mbgLtNgRefclockState', 1)
```
for each element of the list?  I think that would save a lot of needless imports and typing in Agent code.  You could just promote each tuple to a ObjectType if it's not already an Object Type.

Reply:
I like the suggestion of accepting the tuple. I probably wrote it this way originally just sticking to the structure that was in the pysnmp doc examples. I would sort of lean towards just accepting the tuples, as it likely would make the documentation more clear. However, I've written it so that it currently accepts both, promoting the tuples to ObjectType if they aren't already like you suggested.